src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogViewModel.java

```diff
@@ -0,0 +1,359 @@
+                                                                                           Localization.lang("Cancel"));
+                if (!overwriteFilePressed) {
+                    // tbFolder.requestFocus();
+                    return;
+                }
+            }
+        }
+
+        loading.set(true);
+
+        try {
+            SharedDatabaseUIManager manager = new SharedDatabaseUIManager(frame);
+            BasePanel panel = manager.openNewSharedDatabaseTab(connectionProperties);
+            setPreferences();
+
+            if (!folder.getValue().isEmpty()) {
+                try {
+                    new SaveDatabaseAction(panel, Paths.get(folder.getValue())).runCommand();
+                } catch (Throwable e) {
+                    LOGGER.error("Error while saving the database", e);
+                }
+            }
+
+            return;
+        } catch (SQLException | InvalidDBMSConnectionPropertiesException exception) {
+
+            frame.getDialogService().showErrorDialogAndWait(Localization.lang("Connection error"), exception);
+
+        } catch (DatabaseNotSupportedException exception) {
+            ButtonType openHelp = new ButtonType("Open Help", ButtonData.OTHER);
+
+            Optional<ButtonType> result = dialogService.showCustomButtonDialogAndWait(AlertType.INFORMATION,
+                                                                                      Localization.lang("Migration help information"),
+                                                                                      Localization.lang("Entered database has obsolete structure and is no longer supported.")
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("Click help to learn about the migration of pre-3.6 databases.")
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("However, a new database was created alongside the pre-3.6 one."),
+                                                                                      ButtonType.OK, openHelp);
+
+            result.filter(btn -> btn.equals(openHelp)).ifPresent(btn -> HelpAction.openHelpPage(HelpFile.SQL_DATABASE_MIGRATION));
+            result.filter(btn -> btn == ButtonType.OK).ifPresent(btn -> openSharedDatabase());
+
+        }
+        loading.set(false);
+    }
+
+    private void setPreferences() {
+        prefs.setType(selectedDBMSType.getValue().toString());
```

Comment:
Return the preferences instead of setting a local variable? Makes the flow more logical in my opinion.